app/browser/reducers/windowsReducer.js

```diff
@@ -4,18 +4,276 @@
 
 'use strict'
 
+const path = require('path')
 const appConstants = require('../../../js/constants/appConstants')
 const windowConstants = require('../../../js/constants/windowConstants')
 const windowState = require('../../common/state/windowState')
 const windows = require('../windows')
 const {makeImmutable} = require('../../common/state/immutableUtil')
+const electron = require('electron')
+const BrowserWindow = electron.BrowserWindow
+const firstDefinedValue = require('../../../js/lib/functional').firstDefinedValue
```

Comment:
can we change to this `const {firstDefinedValue} = require('../../../js/lib/functional')`?